### PR TITLE
feat: team and workspace flow

### PR DESCRIPTION
**Pull request by Astitva877**
## Description

This PR contains the  whole team and workspace flow except Teams LHS Panel  
#1065
#1080 

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [x] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [x] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build

## Have you tested locally?

- [ ] 👍 yes
- [ ] 🙅 no, because I am lazy

### CODE REVIEW

**Comment by pratikshinde590** [comment]
Please attach the issue link

File: src/packages/@app/pages/Teams/Teams.ViewModel.old.ts [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+    return function (...args) {
+      /* eslint-disable @typescript-eslint/no-this-alias */
+      const context = this;
+
+      clearTimeout(timerId);
+      timerId = setTimeout(() => {
+        func.apply(context, args);
+      }, delay);
+    };
+  };
```
Comment (itsmdasifraza) [inline src/packages/@app/pages/Teams/Teams.ViewModel.old.ts:43]: this file is not required

**Comment by pratikshinde590** [comment]
Please attach the issue link

**Comment by pratikshinde590** [comment]
Please attach the issue link

File: src/packages/@app/pages/Teams/Teams.ViewModel.old.ts (hunk 1 above)
Comment (itsmdasifraza) [reply src/packages/@app/pages/Teams/Teams.ViewModel.old.ts]: you have introduced 2 new view model, that is not required. just rename the existing one to old and create only 1 view model.

File: src/packages/@teams/features/team-explorer/components/workspace-grid-view/WorkspaceGridView.svelte [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+    DoubleLeftIcon,
+    DoubleRightIcon,
+    LeftIcon,
+    RightIcon,
+  } from "$lib/assets/app.asset";
+  import type { Team } from "$lib/utils/interfaces";
+  import type { WorkspaceDocument } from "@app/database/database";
+  import Button from "@library/ui/button/Button.svelte";
+  import WorkspaceGrid from "$lib/components/dashboard/workspace-grid/WorkspaceGrid.svelte";
+
```
Comment (itsmdasifraza) [inline src/packages/@teams/features/team-explorer/components/workspace-grid-view/WorkspaceGridView.svelte:12]: not required to import this component  from $lib that represents deprecate. please push this component to newer folder structure

File: src/packages/@teams/features/team-explorer/components/workspace-list-view/WorkspaceListView.svelte [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+  import {
+    DoubleLeftIcon,
+    LeftIcon,
+    RightIcon,
+    DoubleRightIcon,
+  } from "$lib/assets/app.asset";
+  import type { TeamDocument } from "@app/database/database";
+  import { calculateTimeDifferenceInDays } from "$lib/utils/workspacetimeUtils";
+  import Table from "../../../../../@deprecate/components/table/Table.svelte";
+  import Rows from "../../../../../@deprecate/components/dashboard/rows/Rows.svelte";
```
Comment (itsmdasifraza) [inline src/packages/@teams/features/team-explorer/components/workspace-list-view/WorkspaceListView.svelte:11]: not required to import this component from $lib that represents deprecate. please push this component to newer folder structure

File: src/packages/@teams/features/team-explorer/components/workspace-grid-view/WorkspaceGridView.svelte (hunk 2 above)
Comment (Astitva877) [reply src/packages/@teams/features/team-explorer/components/workspace-grid-view/WorkspaceGridView.svelte]: Done, although didn't removed the old files as their were using in .old files so will remove them later

File: src/packages/@teams/features/team-explorer/components/workspace-list-view/WorkspaceListView.svelte (hunk 3 above)
Comment (Astitva877) [reply src/packages/@teams/features/team-explorer/components/workspace-list-view/WorkspaceListView.svelte]: same

**Review by itsmdasifraza — APPROVED** [review]

**Review by gc-codes — APPROVED** [review]

**Pull request merged**